add basic mask for name validation

kdv24 opened the pull request:
closes #839 

I've never done much with regex, so I suspect there is a way to make this more elegant, but I need to run out for a bit and won't be able to get back to this for a couple hours.  Ideally, the regex character checking would be for the number of characters in the name, not a set amount I choose.  Because I chose a certain length, the user will see a line after their name until they hit the length of my regex- not ideal.

![image](https://user-images.githubusercontent.com/6107653/65352274-1891c480-db9f-11e9-81e5-b1d17a5ace56.png)

I checked and this allows names to have hyphens, spaces, and upper and lower case letters only.  I did not do anything to allow for ~, ^, accents, or umlauts(sp?).  

I'm happy to spend more time on this later today, or would be delighted if someone knew how to fix the length issue now.

kdv24 committed:
add basic mask for name validation

punkael commented on app/src/Pages/Portal/Contributions/Utils/ContributionsFields.js:
```diff
@@ -366,14 +367,14 @@ export const fields = {
     label: "Contributor's First Name",
     section: FormSectionEnum.CONTRIBUTOR,
-    component: TextField,
+    component: NameField,
```
I would use yup and this would match: start with a letter, end with a letter, and allow single spaces, dashes, and apostrophes and any number of chars in the middle (including international chars):
 /^[\p{L}'][ \p{L}'-]*[\p{L}]$/u 

firstName: {
    label: "Contributor's First Name",
    section: FormSectionEnum.CONTRIBUTOR,
    component: TextField,
    validation: Yup.string('Name must only contain name characters')
      .matches(/^[\p{L}'][ \p{L}'-]*[\p{L}]$/u, {
        excludeEmptyString: true,
      })
      .nullable(),
  },

kdv24 committed:
update validation mask

- provide feedback to user
- remove unnecessary input prevention added by NameField
- trim trailing whitespace when map to db

punkael committed:
Allow addtional chars for entity names

punkael committed:
Merge branch 'develop' into mask_invalid_chars-GH-839
